src/dune_rules/dune_package.ml

```diff
@@ -192,7 +190,22 @@ module Lib = struct
        let+ synopsis = field_o "synopsis" string
        and+ loc = loc
        and+ modes = field_l "modes" Lib_mode.decode
-       and+ kind = field "kind" Lib_kind.decode
+       and+ kind =
+         let* kind = field "kind" Lib_kind.decode in
+         let+ virtual_ = field_b "virtual" in
+         match kind with
+         | (Normal | Virtual) when virtual_ ->
+           (* Backward compatible support for dune-project files that
+              that include the [(virtual)] field. )
+              TODO: can we migrate this instead,
+              If not, how do we eventually depricate it? *)
```

Comment:
The customary thing to do would be to look for a `kind` field and use that. Otherwise, look for a `virtual` field.